### ISSUE TITLE
flake8==5.0.4 not compliant with autopep8==2.0.1

### ISSUE THREAD

**Issue by sebastien-genete** [PyCQA/flake8#1769]
### how did you install flake8?

```console
pip install -r requirements-dev.txt

or 

./venv/bin/pip-compile requirements-dev.txt --output-file requirements-ci-dev.txt
./venv/bin/pip install --disable-pip-version-check -r requirements.txt -r requirements-ci-dev.txt 

My file requirements-dev.txt contains:

autodoc-pydantic
autopep8 #==2.0.0 # Temporary fix: 2.0.1 requires pycodestyle >=2.10 but flake8==5.0.4 requiert pycodestyle <2.10...
deprecated
flake8
flake8-awesome
flake8-blind-except
flake8-deprecated
flake8-docstrings
flake8-meiqia
flake8-mutable
flake8-pep3101
flake8-pie
flake8-quotes
flake8-rst-docstrings
flake8-string-format
ipdb
jupyter
jupyter_contrib_nbextensions
matplotlib
myst_parser
pandas-vet
pandera
pipreqs
pre-commit
psycopg2-binary
py
pycodestyle
pycodestyle_magic
pylint
pylint-venv
pytest
pytest-cov
pytest-flake8
pytest-lazy-fixture
pytest-sugar
python-decouple
pyvis
sphinx
sphinx_click
sphinx_rtd_theme
tuna
twine
```
```


### unmodified output of `flake8 --bug-report`

```json
Platform: linux
```

#### what I expected to happen
Since the 15th of dec 2022, I can't install flake8
```
WARNING: using legacy resolver is deprecated and will be removed in future versions. The default resolver will be change to 'backtracking' in 7.0.0 version. Specify --resolver=backtracking to silence this warning.
Using legacy resolver. Consider using backtracking resolver with `--resolver=backtracking`.
Could not find a version that matches pycodestyle<2.10.0,>=2.10.0,>=2.9.0 (from -r requirements-dev.txt (line 26))
Tried: 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.2.0, 2.2.0, 2.3.0, 2.3.0, 2.3.1, 2.3.1, 2.4.0, 2.4.0, 2.5.0, 2.5.0, 2.6.0, 2.6.0, 2.7.0, 2.7.0, 2.8.0, 2.8.0, 2.9.0, 2.9.0, 2.9.1, 2.9.1, 2.10.0, 2.10.0
Skipped pre-versions: 1.8.0.dev0, 1.8.0.dev0, 2.0.0a1, 2.0.0a1, 2.6.0a1, 2.6.0a1
There are incompatible versions in the resolved dependencies:
  pycodestyle (from -r requirements-dev.txt (line 26))
  pycodestyle (from flake8-print==5.0.0->flake8-awesome==1.3.0->-r requirements-dev.txt (line 5))
  pycodestyle<2.10.0,>=2.9.0 (from flake8==5.0.4->-r requirements-dev.txt (line 4))
  pycodestyle>=2.10.0 (from autopep8==2.0.1->-r requirements-dev.txt (line 2))
```

**Comment by sigmavirus24** [comment]
Why aren't you using flake8 6.0? That should resolve this for you

**Comment by sebastien-genete** [comment]
I am not explicitly using flake8 5.0.4.
An idea why flake8 6.0 is not installed with my requirements-dev.txt ?

**Comment by sigmavirus24** [comment]
The output you pasted seems to think that 5.0.4 is pinned in `requirements-dev.txt`, quoth:

```
There are incompatible versions in the resolved dependencies:
  pycodestyle (from -r requirements-dev.txt (line 26))
  pycodestyle (from flake8-print==5.0.0->flake8-awesome==1.3.0->-r requirements-dev.txt (line 5))
  pycodestyle<2.10.0,>=2.9.0 (from flake8==5.0.4->-r requirements-dev.txt (line 4))
  pycodestyle>=2.10.0 (from autopep8==2.0.1->-r requirements-dev.txt (line 2))
```

It seems to imply that you have (or something has) `flake8==5.0.4` and `autopep8==2.0.1`. The way that output prints would imply that `flake8==5.0.4` is coming from `requirements-dev.txt (line 4)`. I can't help you any further because this is not reproducible for me.

All I can guess is that something in 

```
flake8-awesome
flake8-blind-except
flake8-deprecated
flake8-docstrings
flake8-meiqia
flake8-mutable
flake8-pep3101
flake8-pie
flake8-quotes
flake8-rst-docstrings
flake8-string-format
```

Has an upper bound on `<6.0` or something, but `flake8-docstrings` is the only one I can assure you does not.
